code/SubsitesVirtualPage.php

```diff
@@ -204,7 +204,15 @@ public function init(){
 		Subsite::$disable_subsite_filter = true;
 		
 		parent::init();
-		
+		// we also need to load the originally referenced page controller and fire it's init() method
+		$original_class = $this->CopyContentFrom()->ClassName;
+		// dont' run init() when page is newly created and does not have an original page assigned to it yet
+		if($original_class != 'SiteTree') {
+			$controller_class = $this->CopyContentFrom()->ClassName . '_Controller';
+			$cc = New $controller_class; // manually instantiate the class to stop 'method on non-object' errors
```

Comment:
Shouldn't this be `$cc = ModelAsController::controller_for($this->CopyContentFrom())`?